src/fu-main.c

```diff
@@ -562,6 +562,20 @@ g_ptr_array_find (GPtrArray *haystack, gconstpointer needle, guint *index_)
 }
 #endif
 
+static gint
+fu_main_install_task_sort_cb (gconstpointer a, gconstpointer b)
+{
+	FuInstallTask *task_a = *((FuInstallTask **) a);
+	FuInstallTask *task_b = *((FuInstallTask **) b);
+	FuDevice *device_a = fu_install_task_get_device (task_a);
+	FuDevice *device_b = fu_install_task_get_device (task_b);
+	if (fu_device_get_order (device_a) < fu_device_get_order (device_b))
+		return -1;
+	if (fu_device_get_order (device_a) > fu_device_get_order (device_b))
+		return 1;
+	return 0;
```

Comment:
I think if the order is found to be the same you should output something as a warning.

Reply:
Agreed.

Reply:
Maaaaybe not. What happens if you have two unifying receivers (no relationship to each other) both needing the same firmware update? They both then have the same order and you get a warning. In practice it doesn't matter how you order them.